src/pages/repositories/index.js

```diff
@@ -0,0 +1,158 @@
+import React, { Component } from "react";
+
+const FILTER_TYPE_MAP = {
+  All: () => true,
+  Public: (repo) => !repo.private,
+  Private: (repo) => repo.private,
+  Fork: (repo) => repo.fork,
+};
+const FILTER_TYPE_NAMES = Object.keys(FILTER_TYPE_MAP);
+
+export default class Repository extends Component {
+  state = {
+    reposData: [],
+    originalReposData: [],
+    repoName: "",
+    repoDescription: "",
+    private: false,
+    fork: false,
+    updateDate: "",
+    isEditMode: false,
+    star: 0,
+    search: "",
+    filterByType: "All",
+    filterByLanguage: "",
+  };
+
+  componentDidMount = () => {
+    this.fetchReposData();
+  };
+
+  fetchReposData = async () => {
+    const response = await fetch("https://api.github.com/users/saritadc/repos");
+    const data = await response.json();
+
+    this.setState({
+      reposData: data,
+      originalReposData: data,
+    });
+  };
+
+  handleOnFilter = (event) => {
+    this.setState({ [event.target.id]: event.target.value });
+  };
+
+  getLanguageName(arrayList, comparisonKey) {
```

Comment:
while you use miltiple .map .filter in same array please keep comment above function that would define its role

Reply:
Name: Document This
Id: oouo-diogo-perdigao.docthis
Description: Automatically generates detailed JSDoc comments in TypeScript and JavaScript files.
Version: 0.8.2
Publisher: oouo-diogo-perdigao
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=oouo-diogo-perdigao.docthis


you can use this extension for providing function description
example: 
![image](https://user-images.githubusercontent.com/22271897/120759906-3e0d6000-c533-11eb-9937-f85f80f241bf.png)

shorthand to implement this extension is; after u install it just fo `/** */` and it would automatically create those stars.